projects/plugins/boost/app/rest-api/endpoints/List_Source_Providers.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Create a new request for source providers.
+ *
+ * Handler for GET '/list-source-providers'.
+ */
+
+namespace Automattic\Jetpack_Boost\REST_API\Endpoints;
+
+use Automattic\Jetpack_Boost\Lib\Critical_CSS\Source_Providers\Source_Providers;
+use Automattic\Jetpack_Boost\REST_API\Contracts\Endpoint;
+
+class List_Source_Providers implements Endpoint {
+
+	public function request_methods() {
+		return \WP_REST_Server::READABLE;
+	}
+
+	public function response( $_request ) {
+		$providers = new Source_Providers();
+		return rest_ensure_response( $providers->get_provider_sources() );
+	}
+
+	public function permissions() {
+		return array();
+	}
```

Comment:
If we set `permissions` to `Signed_With_Blog_Token` ([example](https://github.com/Automattic/jetpack/blob/f98e248acf2ef55dee12a5558df9ab1401c638a0/projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php#L113)) then we can only allow access from WordPress.com.